pkg/util/net.go

```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
 
 	return status
 }
+
+func RetryTCPConnection(host string, port string) int {
+	status := 0
+	err := wait.Poll(5*time.Second, 120*time.Second, func() (bool, error) {
+		timeout := 3 * time.Second
```

Comment:
can you check what's the default timeout in `net/http`? 3 seconds could be too short

Reply:
The default timeout is no timeout so it will keep trying infinitely. 
https://github.com/golang/go/issues/24138

Increasing it to `30 * time.Second`, I see it is the recommended timeout -> https://pkg.go.dev/net/http